src/Model/Storage.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Kirby\Model;
+
+use Kirby\Data\Data;
+use Kirby\Exception\LogicException;
+use Throwable;
+
+abstract class Storage
+{
+	public function __construct(
+		protected Model $model,
+	) {}
+
+	abstract protected static function contentFile($meta): string;
+
+	public function children(): array
+	{
+		throw new LogicException('This model does not have children');
+	}
+
+	public function files(): array
+	{
+		throw new LogicException('This model does not have files');
+	}
+
+	abstract public static function find(string $class, string $identifier): Model|null;
+
+	public function read(): array
+	{
+		try {
+			return Data::read($this->contentFile($this->model->meta()));
+		} catch (Throwable $e) {
+			return [];
+		}
+	}
+
+	public function changeMeta(array $meta): ModelMeta
+	{
+		$newMeta = clone $this->model->meta();
+
+		foreach ($meta as $key => $value) {
+			$newMeta->$key = $value;
+		}
+
+		foreach (array_keys($meta) as $key) {
+			$this->{'change' . ucfirst($key)}($this->model->meta(), $newMeta);
+		}
+
+		return $newMeta;
+	}
+
+	public function update(array $data): void
```

Comment:
`Suggestion`: We could use the opportunity to bring the two update methods in line: Maybe `updateContent` and `updateMeta`? We have the same issue with many other methods like `read` and `write`. Maybe we need a totally different solution to keep the meta and content spheres separate.

Reply:
You are totally right. I didn't want to throw all our storage stuff over board already and tried to keep close to what we have, but it does not feel ideal indeed. We could also really think about a separation between Content Storage, Meta Storage and for files there could even be a Source Storage. But we also have to be careful that it does not get too crazy complex.